sklearn/preprocessing/data.py

```diff
@@ -1946,3 +1957,149 @@ def transform(self, X):
         """
         return _transform_selected(X, self._transform,
                                    self.categorical_features, copy=True)
+
+
+def _boxcox(x, lambda_x=None):
+    if lambda_x is None:
+        x, lambda_x = stats.boxcox(x, lambda_x)
+        return x, lambda_x
+    else:
+        x = stats.boxcox(x, lambda_x)
+        return x
+
+
+def boxcox(X, copy=True):
```

Comment:
Every time I look again at this PR, I think the function version is a bad idea. it will be misused and cause test data leakage.